charms/istio-pilot/src/charm.py

```diff
@@ -138,16 +138,48 @@ def handle_default_gateway(self, event):
         Side effect: self.handle_ingress() is also invoked by this handler as ingress objects
         depend on the default_gateway
         """
-        t = self.env.get_template('gateway.yaml.j2')
-        gateway = self.model.config['default-gateway']
-        manifest = t.render(name=gateway, app_name=self.app.name)
+        # Clean-up resources
         self._delete_existing_resource_objects(
             resource=self.gateway_resource,
             labels={
                 "app.juju.is/created-by": f"{self.app.name}",
                 "app.{self.app.name}.io/is-workload-entity": "true",
             },
         )
+
+        t = self.env.get_template('gateway.yaml.j2')
+        gateway = self.model.config['default-gateway']
+        secret_name = (
+            f"{self.app.name}-gateway-secret"
+            if self.model.config["ssl-crt"] and self.model.config["ssl-key"]
+            else None
+        )
+        manifest = None
+        if secret_name:
+            manifest = t.render(
+                name=gateway,
+                secret_name=secret_name,
+                ssl_crt=self.model.config["ssl-crt"],
```

Comment:
Do we need `ssl_crt` and `ssl_key` for rendering `gateway.yaml.j2`? If so, they are missing in the yaml file, if not, we can ditch them here.